src/core/CoolBoardSensors.cpp

```diff
@@ -206,15 +206,21 @@ float CoolBoardSensors::readSoilMoisture() {
   digitalWrite(MOISTURE_SENSOR_PIN, LOW);
   digitalWrite(ANALOG_MULTIPLEXER_PIN, HIGH);
   delay(2000);
-
-  // read moisture sensor value
-  int val = analogRead(A0);
-
-  if (val >= 891) {
-    val = 890;
+  int val = 0;
+  // oversample value 64 times for stable readings
+  for (int i = 0; i < 64; i++) {
+    delay(10);
+    val = val + analogRead(A0);
+    delay(10);
   }
-
-  float result = (float)map(val, 0, 890, 0, 100);
+  // divide by 64 to get the average
+  val=val/64;
+  float x_deux = 0.00027;
+  float x_un = -0.13;
+  float x_zero= -4;
+  float result = x_deux*(float)(val*val) + x_un*(float)val+ x_zero;
+  if( result > (float)100) result = (float)100 ;
+  if( result <(float)0 ) result = (float)0 ;
```

Comment:
there are a few problems with this function:
-  `x_un` *et al* are not functions but constants. The should be defined using `#define MY_CONSTANT` to stay consistent with the rest of the code.
- `x_un` is not a very meaningful name
- the way you compute result is not very clear. What are you trying to achieve ? What is the name of the mathematical tools you are using here ? Why are you using it ? Those should be understandable just by reading your code. If you were asked to solve a problem in math or physices, you would never answer just by writing the computation, but surround it with something like "by theorem X" (math) or a reference the model being used (physics), or any relevant information clarifying your intent.

The same applies to code : your intent should be clearer. However, we do not encourage comments for a number of reasons. We favor naming things such as variables and function names so that they respectively explain what value they hold or what they are achieving